src/server/storyRequestHandlers.js

```diff
@@ -157,28 +202,51 @@ fluid.defaults("sjrk.storyTelling.server.saveStoryHandler", {
  * @param {Boolean} authoringEnabled - a server-level flag to indicate whether authoring is enabled
  */
 sjrk.storyTelling.server.handleSaveStory = function (request, dataSource, authoringEnabled) {
-    if (authoringEnabled) {
-        sjrk.storyTelling.server.saveStoryToDatabase(dataSource, request.req.body, request.events.onSuccess, request.events.onError);
-    } else {
+    if (!authoringEnabled) {
         request.events.onError.fire({
+            statusCode: 403,
             isError: true,
             message: "Saving is currently disabled."
         });
+
+        return;
     }
+
+    var id = request.req.params.id;
+
+    dataSource.get({directStoryId: id}).then(function (response) {
+        if (request.req.session.authorID === response.authorID) {
+            sjrk.storyTelling.server.saveStoryToDatabase(dataSource, request.req.session.authorID, request.req.body, request.events.onSuccess, request.events.onError);
```

Comment:
Could you add a code comment here and at line 226 that this line is to update an existing story while line 226 is to create a new story.